daffodil-japi/src/test/java/org/apache/daffodil/example/TestJavaAPI.java

```diff
@@ -863,7 +849,6 @@ public void testJavaAPI17() throws IOException, InvalidUsageException, ClassNotF
         assertTrue(res.isError());
         assertFalse(res.isProcessingError());
         assertTrue(res.isValidationError());
-        assertTrue(res.location().isAtEnd());
```

Comment:
Is it worth adding ``res.location().bitPos0b == lengthOfData`` comparison? We sometimes referene this file for examples of Java API usage, so showing how to test that we consumed all the data might be useful.

Reply:
Added these checks here and in SAPI.